armi/reactor/converters/uniformMesh.py

```diff
@@ -540,10 +522,6 @@ def setAssemblyStateFromOverlaps(
                 destBlock, updatedDestVals.values(), updatedDestVals.keys()
             )
 
-            UniformMeshGeometryConverter._applyCachedParamValues(
-                destBlock, blockParamNames, cachedParams
-            )
-
```

Comment:
My understanding of this change is as follows:

- Before, we were clearing all of the block parameters on the new, converted assembly, while storing some of them temporarily in a "cache". Then, we were mapping the specified parameters to the new assembly, and the cached parameters are then used to put the other parameters back onto the blocks.
- Now, we just avoid clearing the parameters in the first place, so that the caching of parameters isn't needed for anything. All parameters that are not specified specifically to be mapped are just left as their initial values?

This seems like potentially a source of great speedup. But it makes me wonder, are we losing something by changing that flow? It seems to me that an expensive step, like clearing all the parameters in the first place, must have been done for a good reason. But I cannot see why that reason might have gone away now...

Reply:
My understanding of the caching is basically the same as yours.

As for whether the caching is necessary, I would break it down into the two directions:

- Non-uniform source, uniform destination assembly: When the destination assembly is the uniform assembly, it's being created from the source assembly, so it shouldn't have any pre-existing parameters that need to be cached.
- Uniform source, non-uniform destination: When the destination assembly is the non-uniform assembly, parameter caching would ensure that we don't erase any of the parameters in `blockParamNames` on the non-uniform assembly by overwriting with null values from the source assembly. But, if we have defined the `blockParamNames` to be mapped back correctly, then we should only be mapping back parameters that have been set on the uniform assembly. If this `setAssemblyStateFromOverlaps` is trying to map back parameters that haven't been set, then either the `_setParamsToUpdate` was not defined correctly or maybe some analysis performed on the uniform reactor failed. `_applyCachedParamValues` would paper over that error by just retaining whatever value the non-uniform assembly already had for that parameter, but this would create a "silent" bug that would be very difficult to track down (e.g., you have a `b.p.power` in the database, but you have no way of verifying whether it was mapped back correctly or just retained by caching).

That's my take on it, but it's complicated and I'm definitely curious to hear the opinions of others. I think the short summary is that parameter caching would not be performing any detectable function when everything is working correctly. It would only act when there is a bug in the code or an error during execution, and it would effectively cover up the error without giving any warning or error message.

Reply:
My major concern with preserving the old, caching solution is backwards compatibility for our downstream projects.

But it doesn't appear well enough documented that I'm thrilled with having to support it forever. Michael's above description matches what I see in the code. But the original developer could have saved us several hours of digging if it had been better documented.

Let's do better!